src/lib/icon/icon-registry.ts

```diff
@@ -327,14 +327,24 @@ export class MdIconRegistry {
    */
   private _extractSvgIconFromSet(iconSet: SVGElement, iconName: string): SVGElement {
     const iconNode = iconSet.querySelector('#' + iconName);
+
     if (!iconNode) {
       return null;
     }
+
     // If the icon node is itself an <svg> node, clone and return it directly. If not, set it as
     // the content of a new <svg> node.
-    if (iconNode.tagName.toLowerCase() == 'svg') {
+    if (iconNode.tagName.toLowerCase() === 'svg') {
       return this._setSvgAttributes(iconNode.cloneNode(true) as SVGElement);
     }
+
+    // If the node is a <symbol>, it won't be rendered so we have to convert it into <svg>. Note
```

Comment:
We're not doing this for all `symbol` elements, but rather for the one that the user is referring to via an id. I think it's safe to assume that if there's a `<symbol id="something">` and the user is referring to it via `iconSet:something`, it would be expected that the symbol would be rendered.